aiida_quantumespresso/utils/restart.py

```diff
@@ -1,451 +1,57 @@
-            'but I got {} instead'.format(parent_calc.__class__.__name__)
-        )
-
-    if not force_restart and not parent_calc.is_finished_ok:
-        raise exceptions.InputValidationError(
-            'Calculation to be restarted must be finshed ok. Otherwise, use the force_restart flag'
-        )
-
-    inputs = parent_calc.get_incoming(link_type=LinkType.INPUT_CALC)
-    code = inputs.get_node_by_label('code')
-    qpoints = inputs.get_node_by_label('qpoints')
-
-    old_inp_dict = inputs.get_node_by_label('parameters').get_dict()
-    # add the restart flag
-    old_inp_dict['INPUTPH']['recover'] = True
-    inp_dict = Dict(dict=old_inp_dict)
-
-    try:
-        remote_folder = parent_calc.get_outcoming(node_class=RemoteData, link_type=LinkType.CREATE).one().node
-    except ValueError:
-        raise InputValidationError(
-            'No or more than one output RemoteData found in calculation {}'.format(parent_calc.pk)
-        )
-
-    builder = parent_calc.__class__.get_builder()
-
-    # Set the same options
-    # TODO: Implement this in a way, in AiiDA, to allow to get the values in a single call,
-    #       returning either only those actually set, or all with their defaults,
-    #       and here just call that method (to re-set those actually set).
-    for option in builder.options._valid_fields:
-        option_val = getattr(parent_calc, 'get_{}'.format(option))()
-        if option_val:  # Skip None, empty strings, and empty dicts/lists
-            setattr(builder.options, option, option_val)
-
-    builder.label = parent_calc.label
-    builder.description = '[Restart of {} {}]\n{}'.format(
-        parent_calc.__class__.__name__, parent_calc.uuid, parent_calc.description
-    )
+        builder.pop('parent_folder', None)
 
-    # set the parameters, and the (same) code and q-points
-    builder.parameters = Dict(inp_dict)
-    builder.code = code
-    builder.qpoints = qpoints
+    # Update the parameters to set the correct restart flag
+    parameters = builder.parameters.get_dict()
 
-    try:
-        old_settings_dict = inputs.get_node_by_label('settings').get_dict()
-    except KeyError:
-        old_settings_dict = {}
-    if parent_folder_symlink is None:
-        parent_folder_symlink = parent_calc._default_symlink_usage
-    # Always set if it was already set. Otherwise, if it wasn't set, just set it if it's not the default
-    if ('PARENT_FOLDER_SYMLINK' in old_settings_dict or parent_folder_symlink != parent_calc._default_symlink_usage):
-        old_settings_dict['PARENT_FOLDER_SYMLINK'] = parent_folder_symlink
+    if node.process_class in (CpCalculation, PwCalculation):
+        parameters.setdefault('CONTROL', {})['restart_mode'] = 'from_scratch' if from_scratch else 'restart'
+    elif node.process_class is PhCalculation:
+        parameters.setdefault('INPUTPH', {})['recover'] = True
+    elif node.process_class is NebCalculation:
+        parameters.setdefault('PATH', {})['restart_mode'] = True
```

Comment:
From the QE docs, this is supposed to be the string `'restart'` for NEB. Unless you tried it to run it and found otherwise.

Reply:
idem